server/src/main/java/org/cloudfoundry/identity/uaa/authentication/UaaLoginHint.java

```diff
@@ -32,6 +35,11 @@ public UaaLoginHint(String origin) {
         this.origin = origin;
     }
 
+    public UaaLoginHint(String origin, String idToken) {
```

Comment:
Can you add the id_token_hint to UaaAuthentication or UaaAuthenticationDetails please, see https://github.com/cloudfoundry/uaa/pull/3053 because this class is used to validate the request ... it is used to keep some client informations but it should not contain authentication information

Reply:
OK I added the id_token_hint to the UaaAuthentication. New commit is shown below.